### PR TITLE
Add feature for auto search of upload method config for custom targets

### PR DESCRIPTION
**Pull request by JohnK1987**
### Summary of changes 
This PR is in context of many discussions and I hope everyone will find its option.
* [#233](https://github.com/mbed-ce/mbed-os/pull/233)
* https://github.com/mbed-ce/mbed-os/discussions/112

In current state Mbed CE build system auto searches only for upload method config for regular targets. In case of custom targets is expected the config will be defined via top level cmakelist.

```
// this is example how look current structure of a project with a custom target
./Project_Name/
│
├── mbed-os/
├── Build/
├── Custom_targets/
│   ├── YOUR_TARGET/
│   │	└──  somefiles.xxx
│   ├── ANOTHER_TARGET/
│   │	└──  somefiles.xxx
│   └──	CMakeLists.txt
│
├── CMakeLists.txt
├── mabed_app.json
├── custom_targets.json
└── main.cpp
```
This PR brings the build system will also search for an upload method config in `custom_targets/upload_method_cfg/` folder (it mirrors same logic from `mbed-os/targets/upload_method_cfg/` folder) and also brings an option to change default expected path via top lvl cmakelist.

```
// this is example how will look new expected structure of a project with a custom target
./Project_Name/
│
├── mbed-os/
├── Build/
├── Custom_targets/
│   ├── YOUR_TARGET/
│   │   └──  somefiles.xxx
│   ├── ANOTHER_TARGET/
│   │   └──  somefiles.xxx
│   │
│   ├── upload_method_cfg/
│   │   ├──  YOUR_TARGET.cmake
│   │   └──  ANOTHER_TARGET.cmake
│   └──	CMakeLists.txt
│
├── CMakeLists.txt
├── custom_targets.json
├── mabed_app.json
└── main.cpp
```

We will have an official automated solution how to deal with upload method config file for custom targets. But also a solution for anyone who want to go its own way.

#### Migration actions required 

N/A

### Documentation 
How will this merged I can update [custom target example 
](https://github.com/mbed-ce/mbed-ce-custom-targets)  as soon as possible.

----------------------------------------------------------------------------------------------------------------
### Pull request type 

    [x] Patch update (Bug fix / Target update / Docs update / Test update / Refactor)
    [] Feature update (New feature / Functionality change / New API)
    [] Major update (Breaking change E.g. Return code change / API behaviour change)

----------------------------------------------------------------------------------------------------------------
### Test results 

    [x] No Tests required for this change (E.g docs only update)
    [] Covered by existing mbed-os tests (Greentea or Unittest)
    [] Tests / results supplied as part of this PR
    
    
----------------------------------------------------------------------------------------------------------------
### Reviewers 

@multiplemonomials 

----------------------------------------------------------------------------------------------------------------

### CODE REVIEW

File: tools/cmake/app.cmake [hunk 1]
```diff
@@ -75,11 +75,30 @@ endif()
 # Load upload method configuration defaults for this target.
 # Loading the settings here makes sure they are set at global scope, and also makes sure that
 # the user can override them by changing variable values after including app.cmake.
-set(EXPECTED_MBED_UPLOAD_CFG_FILE_PATH targets/upload_method_cfg/${MBED_TARGET}.cmake)
-if(EXISTS ${CMAKE_CURRENT_LIST_DIR}/../../${EXPECTED_MBED_UPLOAD_CFG_FILE_PATH})
-    message(STATUS "Mbed: Loading default upload method configuration from ${EXPECTED_MBED_UPLOAD_CFG_FILE_PATH}")
-    include(${CMAKE_CURRENT_LIST_DIR}/../../${EXPECTED_MBED_UPLOAD_CFG_FILE_PATH})
+#
+# default expected paths
+set(EXPECTED_CUSTOM_UPLOAD_CFG_FILE_PATH ${CMAKE_SOURCE_DIR}/custom_targets/${MBED_TARGET}/${MBED_TARGET}.cmake)
```
Comment (multiplemonomials) [inline tools/cmake/app.cmake:80]: Could we change this to `${CMAKE_SOURCE_DIR}/custom_targets/upload_methods/${MBED_TARGET}.cmake` to mirror the structure inside Mbed?

**Review by multiplemonomials — COMMENTED** [review]
One comment otherwise looks good!

File: tools/cmake/app.cmake (hunk 1 above)
Comment (JohnK1987) [reply tools/cmake/app.cmake]: Updated!

**Review by multiplemonomials — APPROVED** [review]
Looks good, thanks for the fix!

**Pull request merged**